compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -52,7 +52,7 @@ impl AllocBytes for Box<[u8]> {
     }
 
     fn zeroed(size: Size, _align: Align) -> Option<Self> {
-        let bytes = Box::<[u8]>::try_new_zeroed_slice(size.bytes_usize()).ok()?;
+        let bytes = Box::<[u8]>::try_new_zeroed_slice(size.bytes().try_into().ok()?).ok()?;
```

Comment:
In case `size` does not fit into a `usize` (which is totally possible when building for a 64bit target on a 32bit host), this avoids another ICE.